### PR TITLE
feat(VTextField): add `counter` scoped slot

### LINKED ISSUE #9548: [Feature Request] Add slot for (.v-counter) in inputs like (v-text-field)
### Proposed solution
```vue
<template v-slot:counter="{ current_length, max }">
  <custom-element>
    {{ current_length }}
    <custom-separator></custom-separator>
    {{ max }}
  </custom-element>
</template>
```
> Developers might also use this slot to display some messages in the other corner beside the (**.v-messages**) slot.

Thank you :)

### PR DESCRIPTION
**Pull request by KaelWD**
resolves #9548

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

### CODE REVIEW

File: packages/vuetify/src/components/VTextField/VTextField.ts [hunk 1]
```diff
@@ -312,14 +312,14 @@ export default baseMixins.extend<options>().extend({
...
-        },
-      })
+      const props = {
+        dark: this.dark,
+        light: this.light,
+        max,
+        value: this.computedCounterValue,
+      }
+
+      return this.$scopedSlots.counter?.({ props }) ?? this.$createElement(VCounter, { props })
```
Comment (johnleider) [inline packages/vuetify/src/components/VTextField/VTextField.ts:322]: Should we use the getSlot helper fn?

**Review by johnleider — COMMENTED** [review]
Looks good, only one comment regarding process.

File: packages/docs/src/examples/v-text-field/prop-counter.vue [hunk 2]
```diff
@@ -35,10 +35,11 @@
...
-            :rules="rules"
-            counter="25"
-            filled
-            label="Filled"
+            :rules="wordsRules"
+            counter="5"
+            hint="This field counts words instead of characters"
+            label="Custom counter from prop"
+            :counter-value="v => v.trim().split(' ').length"
           ></v-text-field>
```
Comment (shamilfrontend) [inline packages/docs/src/examples/v-text-field/prop-counter.vue:43]: `/>` ?

File: packages/docs/src/examples/v-text-field/prop-counter.vue [hunk 3]
```diff
@@ -48,11 +49,15 @@
...
-            label="Outlined"
-            outlined
-          ></v-text-field>
+            :rules="wordsRules"
+            counter="5"
+            hint="This field counts words instead of characters"
+            label="Custom counter from slot"
+          >
+            <template v-slot:counter="{ props }">
+              <v-counter v-bind="props" :value="title.trim().split(' ').length"></v-counter>
```
Comment (shamilfrontend) [inline packages/docs/src/examples/v-text-field/prop-counter.vue:58]: `/>`?

File: packages/docs/src/examples/v-text-field/prop-counter.vue (hunk 2 above)
Comment (Shinigami92) [reply packages/docs/src/examples/v-text-field/prop-counter.vue:43]:
```suggestion
          />
```

File: packages/docs/src/examples/v-text-field/prop-counter.vue (hunk 3 above)
Comment (Shinigami92) [reply packages/docs/src/examples/v-text-field/prop-counter.vue:58]:
```suggestion
              <v-counter v-bind="props" :value="title.trim().split(' ').length"/>
```

File: packages/docs/src/examples/v-text-field/prop-counter.vue (hunk 2 above)
Comment (KaelWD) [reply packages/docs/src/examples/v-text-field/prop-counter.vue:43]: Self-closing tags don't work in DOM templates so we don't use them in examples.

**Pull request merged**